src/interpreter/initialization.rs

```diff
@@ -124,19 +129,20 @@ impl<S> Interpreter<S> {
             let fee = bytes.checked_add(gas).ok_or(ValidationError::ArithmeticOverflow)?;
 
             // subtract total fee from base asset balance
-            *base_asset_balance =
+            let deducted_balance =
                 base_asset_balance
                     .checked_sub(fee)
                     .ok_or(ValidationError::InsufficientFeeAmount {
                         expected: fee,
                         provided: *base_asset_balance,
                     })?;
+
+            *base_asset_balance = deducted_balance;
         }
 
-        // reduce free balances by coin and withdrawal outputs
+        // reduce free balances by coin
         for (asset_id, amount) in self.tx.outputs().iter().filter_map(|output| match output {
             Output::Coin { asset_id, amount, .. } => Some((asset_id, amount)),
-            Output::Message { .. } => todo!(),
```

Comment:
Wait, message outputs still count as "withdrawals," or rather, they should still reduce the free balance. What is this change doing? Is the output amount accounted for?

Reply:
OutputMessage amount is 0 or the whole transaction is invalid.

Reply:
Ah right they're variable.